ui/ui-guidelines.md

```diff
@@ -86,62 +236,47 @@ are reflected in how the codebase is organized.
-  generally more time consuming to run and more prone to flakiness. That being said,
-  each large feature/project should include at least one e2e test to ensure the most
+- Try to use the ARIA role-based selectors when writing tests and don’t be afraid to
+  go back and modify components to make them accessible to those selectors in tests.
+  This has the advantage of adding at least a modicum of accessibility to the
+  application as we build it. Using a “test id” should be the absolute last resort.
+
+### End-to-end and browser testing
+
+End-to-end and browser tests are written using [Playwright](https://playwright.dev/). These tests are significantly more time-consuming to run, so they account for a small portion of our test suite. They should focus on [smoke testing](<https://en.wikipedia.org/wiki/Smoke_testing_(software)>) critical user flows and testing use cases that are difficult to cover using other tools (e.g. drag and drop interactions).
+
+- Tests live in the `ui/e2e` directory. See the [README](./e2e) in that directory for additional guidance.
+- Each large feature should include at least one e2e test to ensure the most
   common flow works.
 
-### Build Process
+### Visual testing
 
-- We use webpack to bundle our application, our config is heavily inspired by Create
-  React App, but ejected because we had the need to configure it to support specific
-  requirements for our project
-- Eslint enforces certain style and correctness rules, these can be ignored on a case
-  by case basis but usually are intended to help you avoid common JS pitfalls
-- Typescript is used to provide a stronger guarantee of correctness across the
-  application. Please avoid using escape hatches like `any`, prefer using `unknown`
-  and type narrowing instead.
-- Prettier is auto formatting all code to avoid bikeshedding on basic stylistic
-  concerns
+- Visualization-heavy components (e.g. charts using `canvas`) should have visual tests using [Happo](https://happo.io/).
+- Visual tests can be generated during the execution of an end-to-end test with Playwright or by creating stories in Storybook. See the [e2e](./e2e) and [storybook](./storybook) readmes for additional guidance.
+
+## Documentation
+
+Code intended for use by library consumers should include documentation using the following tools:
+
+- jsdoc in comments explaining the purpose of components, classes, and functions.
+- jsdoc on types and their properties, especially types that define component props.
+- Stories using storybook for components. Stories autogenerate a significant portion of their documentation using jsdoc comments, so the prior items are important for this to work well.
+
+> ⚠️ Note: this is an aspirational guideline that should impact _new_ development. There is a small amount of documentation for the current codebase, and we are slowly working on expanding it.
 
-## Frontend Best Practices
+## Additional best bractices
 
```

Comment:
I tried to organize most of these into relevant topics. What's left are bits I wasn't sure where to organize them.

Reply:
Makes sense! There is a typo, should be a “p”

Reply:
Oh noooo! That's not a great typo. Thanks for catching that.